File: Two Sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_dict = {}
        for i, x in enumerate(nums):
            num_dict[x] = i
        for i, x in enumerate(nums):
            if target - x in num_dict:
                if i == num_dict[target - x] :
                    continue
                return (i, num_dict[target - x])

File: Two Sum/TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
    	int[] ans = {0, 0};
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        map.clear();
        for(int i=0; i<nums.length; i++) {
        	map.put(nums[i], i);
        }
        for(int i=0; i<nums.length; i++) {
        	if(map.containsKey(target - nums[i])) {
        		ans[0] = i;
        		ans[1] = map.get(target - nums[i]);
        		if(ans[0] == ans[1]) {
        			continue;
        		}
        		break;
        	}
        }
        return ans;
    }
}